src/commands/deploy.rs

```diff
@@ -231,8 +231,8 @@ impl DeployCommand {
                 let source_dir = crate::app_dir(&self.app)?;
                 let fm = assets::collect(files, &source_dir)?;
                 for f in fm.iter() {
-                    let src_path = source_dir.join(&f.src);
-                    let mut r = File::open(src_path)?;
+                    let mut r = File::open(&f.src)
```

Comment:
Hey @FrankYang0529 - I think if we remove line 34, it might introduce a regression in that running deploy from outside of the app directory may not work as expected. Would you mind explaining this change?

Reply:
Hi @michelleN, if we don't update the file path, we may have an error like this https://github.com/fermyon/spin/issues/590#issuecomment-1167611846.

```console
# before updating the file path, but adding with_context to show informative error message.
# let mut r = File::open(&src_path).with_context(|| anyhow!("Cannot open file {}", &src_path.display()))?;

./target/debug/spin deploy --file tests/http/simple-spin-rust/spin.toml --bindle-server=http://localhost:8080/v1 --hippo-server=http://localhost:5309 --hippo-username=admin --hippo-password=password --deploy-existing-bindle
Error: Cannot open file tests/http/simple-spin-rust/tests/http/simple-spin-rust/assets/test.txt

Caused by:
    No such file or directory (os error 2)
```

```console
# after updating the file path, although there is error from hippo, it has been fixed in https://github.com/deislabs/hippo/issues/978
./target/debug/spin deploy --file tests/http/simple-spin-rust/spin.toml --bindle-server=http://localhost:8080/v1 --hippo-server=http://localhost:5309 --hippo-username=admin --hippo-password=password --deploy-existing-bindle
Error: Unable to create Hippo app

Caused by:
    {"type":"https://tools.ietf.org/html/rfc7231#section-6.6.1","title":"An error occurred while processing your request.","status":500}
```